lib/python/test/testsuite/Math/_searchData.py

```diff
@@ -0,0 +1,81 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2022 PIConGPU contributors
+Authors: Mika Soren Voss
+License: GPLv3+
+
+Routines in this module:
+
+searchParameter(parameter:str, directiontype:str = None)
+"""
+
+__all__ = ["searchParameter"]
+
+from .. import Reader
+
+
+def searchParameter(parameter: str, directiontype: str = None, **kwargs):
+    """
+    searches in the data (passed in Data.py as a directory).
+    To avoid errors, usage without directiontype is not recommended
+
+    Input:
+    -------
+
+    parameter :     str
+                    Designation of the parameters. Must match the
+                    designations in the data files.
+
+    directiontype : str, optional
+                    "param", "json", "dat" or "openpmd".
+                    If None, the parameter is searched for in all files
+                    specified in Data.py.
+
+    Raise:
+    -------
+    ValueError:
+        if directiontype not in ["param", "json", "dat", "openpmd", None]
+        if the parameter could not be found
+
+    Return:
+    -------
+    out : float or array
+    """
+
+    if (directiontype not in ["param", "json", "dat", "openpmd"] and
+            directiontype is not None):
+        raise ValueError("directiontype must be either None, param,"
+                         " dat, json, or openpmd")
+
+    if directiontype == "param":
+        result = Reader.paramReader.getValue(parameter)
+    elif directiontype == "json":
+        result = Reader.jsonReader.getValue(parameter)
+    elif directiontype == "dat":
+        result = Reader.dataReader.getValue(parameter, **kwargs)
+    # ToDo: openpmd
+
+    if directiontype is None:
+        try:
+            result = Reader.paramReader.getValue(parameter)
+        except Exception:
+            result = None
+
+        if result is None:
+            try:
+                result = Reader.jsonReader.getValue(parameter)
+            except Exception:
+                result = None
+
+        if result is None:
+            try:
+                result = Reader.dataReader.getValue(parameter)
+            except Exception:
+                result = None
```

Comment:
This is the same as line 60-63, except that it is checked if `result is None`. But since this has been checked before and the `result` is either written or not, this will not be executed except when running again into the `except`.
Did you intend to check for openPMD files here?